### ISSUE TITLE
Cannot build SquashFS image after #1506 was added

### ISSUE THREAD

**Issue by mcassaniti** [systemd/mkosi#1540]
Using the configuration below causes a build failure of "No space left on device". The commit [Do prebuilt initrds by default](https://github.com/systemd/mkosi/pull/1506/commits/7f820b740730fe073d274181d3dd3843c96aa706) from #1506 has introduced a regression.

# Configuration

```shell
$ cat mkosi.conf
[Distribution]
Distribution = ubuntu
Release      = lunar
Repositories = main,universe

[Output]
ImageId        = Rescue
ImageVersion   = 0
Format         = disk
SplitArtifacts = yes
SecureBoot     = yes

CompressOutput = zst

# Make root volatile
KernelCommandLine = systemd.volatile=overlay

[Validation]
Checksum  = yes

[Content]
Autologin = yes
Packages  = ubuntu-minimal,dracut,linux-image-generic,systemd-boot-efi

$ cat mkosi.repart/00-ESP.conf 
[Partition]
Type         = esp
Label        = ESP
Format       = vfat

SizeMinBytes = 128M
SizeMaxBytes = 128M

CopyFiles    = /boot/EFI:/

$ cat mkosi.repart/10-os.conf 
[Partition]
Type           = root
Label          = rescue
Format         = squashfs

Minimize       = best
CopyFiles      = /
ExcludeFiles   = /boot/EFI

VerityMatchKey = os
Verity         = data

$ cat mkosi.repart/11-os-verity.conf 
[Partition]
Type           = root-verity
Label          = rescue

SizeMinBytes   = 50M
SizeMaxBytes   = 50M

VerityMatchKey = os
Verity         = hash
```

**Comment by mcassaniti** [comment]
I do have to ask about regression testing. When would we call the output to be expected from `mkosi` stable and when would we call the configuration file format stable against backward compatibility?

**Comment by DaanDeMeyer** [comment]
> I do have to ask about regression testing. When would we call the output to be expected from mkosi stable and when would we call the configuration file format stable against backward compatibility?

I can't give a date on that. I'm still fixing a bunch of stuff that requires backwards compat breaks. Once we do the next release I'm planning on introducing some kind of backwards compat guarantee but no timeline on when that'll happen.

**Comment by DaanDeMeyer** [comment]
As for the issue, your ESP is too small to fit the prebuilt initrd. This is primarily because we include all kernel modules in the initrd and Debian/Ubuntu do not compress their kernel modules, which means the kernel modules take up 642M.

The real fix here is for Debian/Ubuntu to start using compressed kernel modules like every other distro. But I'm willing to take PRs for an option that disables usage of the extra kernel modules initrd (--kernel-modules-initrd=yes/no) and options to configure which modules to include in the kernel modules initrd (--kernel-modules-initrd-include, --kernel-modules-initrd-exclude). 

If you don't use the kernel modules initrd, you'll have to make sure the kernel modules you need are available in the initrd yourself.

**Comment by mcassaniti** [comment]
Are you telling me I'll have a UKI at around 650M?

**Comment by DaanDeMeyer** [comment]
#1542 will improve the situation a bit.

**Comment by mcassaniti** [comment]
How do I simply use the initrd that is built by the distro again? I can see the options `KernelModulesInitrd=` and `Initrd=` are likely relevant, but how do I extract the right image?

To make things worse, I'm adding a newer version of systemd from the build script which gets included into the initrd, along with some dracut modifications pulled in through `mkosi.extra`.

**Comment by DaanDeMeyer** [comment]
It's pretty trivial to build a custom initrd that is then used in the custom image. See the systems repo for an example (mkosi.presets directory). I still need to document that.

**Comment by DaanDeMeyer** [comment]
Closing this one as the relevant options should be available to reduce initramfs size. Please open a new issue if there are still specific problems
